lib/rules/no-restricted-modules.js

```diff
@@ -106,10 +106,19 @@ module.exports = {
          * @param {ASTNode} node The node to check.
          * @returns {boolean} If the node is a string literal.
          */
-        function isString(node) {
+        function isStringLiteral(node) {
             return node && node.type === "Literal" && typeof node.value === "string";
         }
 
+        /**
+         * Function to check if a node is a string template literal.
+         * @param {ASTNode} node The node to check.
+         * @returns {boolean} If the node is a string template literal.
+         */
+        function isStringTemplateLiteral(node) {
+            return node && node.type === "TemplateLiteral" && node.quasis.length && typeof node.quasis[0].value.raw === "string";
+        }
+
```

Comment:
This could produce false positives on:

```js
/* eslint no-restricted-modules: ["error", "foo"] */

require(`foo${bar}`);
```

`"TemplateLiteral"` nodes always have at least one element in `quasis` and their values are always `string` values.

The correct condition would be:

```js
return node && node.type === "TemplateLiteral" && node.expressions.length === 0;
```

Maybe we could also rename this function to `isStaticTemplateLiteral`?